### ISSUE TITLE
Feature request: MOCHA_GREP environment variable (alternative to --grep)

### ISSUE THREAD

**Issue by jedwards1211** [mochajs/mocha#4232]
**Is your feature request related to a problem or a nice-to-have?? Please describe.**
I've been working with `sequelize` lately and it's got a bunch of package scripts for testing, here are a subset:
```
    "test-docker": "npm run test-docker-unit && npm run test-docker-integration",
    "test-docker-unit": "npm run test-unit",
    "test-docker-integration": "env-cmd $npm_package_options_env_cmd npm run test-integration",
    "test-integration": "mocha --require scripts/mocha-bootload --globals setImmediate,clearImmediate --exit --check-leaks --colors -t 30000 --reporter spec \"test/integration/**/*.test.js\"",
    "test-integration-postgres": "cross-env DIALECT=postgres npm run test-integration",
```
Grepping for tests is really annoying because
1) I have to use a bunch of `--`, like `npm run test-docker-integration -- -- --grep foo`, to pass `--grep` to the actual `mocha` command.  `yarn` doesn't have this problem if you use `yarn` instead of `npm run` in the package scripts, but I don't control that.
2) `npm run test-docker -- -- --grep foo` would only apply the `--grep` to the `test-docker-integration` script, not to the `test-docker-unit` script.

**Describe the solution you'd like**
If I could just do `MOCHA_GREP=foo npm run test-docker` it would be a lot easier.

**Describe alternatives you've considered**
I'm planning to PR some additional scripts to their repo for grepping tests, but I'm sure `sequelize` isn't the only repo that makes grepping difficult like this, so I'd rather have a quick and easy way to do it for future projects I may work on.

**Comment by outsideris** [comment]
Couldn't you use [configuration files](https://mochajs.org/#configuring-mocha-nodejs)?

**Comment by jedwards1211** [comment]
I guess that's true...it would be a bit more of a hassle to edit the grep and avoid committing the file to the repo than if I could just use an environment variable though

**Comment by outsideris** [comment]
I don't agree to add `MOCHA_GREP`. Because if we support mocha flags with environment variable, we should support all flags with environment variable rather than only `--grep`. 
It makes hard to maintain mocha's flags.
